cookbooks/calico/templates/default/compute/bird6.conf.erb

```diff
@@ -44,13 +44,13 @@ protocol direct {
 protocol bgp <%= n[:fqdn].gsub(/[^0-9a-z]/i, '') %> {
```

Comment:
I'm actually wondering if all of this processing should just be in this template.  You can insert arbitrary ruby code in to a template, so you could basically put some stuff in there to work out which IPv6 address to use.  I think this would be easier to implement and it would just involve changing one file.